### DISCUSSION TITLE
DTO and Prisma Schema

### DISCUSSION THREAD

**Discussion by Dsantosjaime** [prisma/prisma#19998] [Q&A]
Hi,

I have this UpdateDTO :

```
export class UpdateUserDto {
    @IsEmail()
    @ApiProperty()
    email?: string;

    @ApiProperty()
    firstname?: string;

    @ApiProperty()
    lastname?: string;

    @IsPhoneNumber()
    @ApiProperty()
    phone?: string;
}

export class UpdateSpecialistDto extends UpdateUserDto {
    @ApiProperty()
    @Type(() => SpecialistDto)
    @ValidateNested()
    specialist?: SpecialistDto;
}
```

And this Schema:

```
model User {
  id         String      @id @default(auto()) @map("_id") @db.ObjectId
  readableId Int?        @unique
  email      String      @unique
 [...]
  specialist Specialist?
}

type Specialist {
  firebaseUid String
  [...]
}
```

The problem:
When I Try:

```
async update(id: string, updateUserDto: UpdateSpecialistDto | UpdateClientDto | UpdateAdminDto) {
        try {
            return this.prisma.user.update({
                where: { id }, data: updateUserDto
            });
        } catch (e) {
            throw new HttpException(e.message, HttpStatus.CONFLICT);
        }
    }
```

It tell me that my field 'firebaseUid' can't be assigned to 'never' because it's a 'string' type.
I guess I'm doing it wrong but can't figure out why.

**Comment by nurul3101** [comment]
Hello @Dsantosjaime 👋 

Can you please share what is the value of `updateUserDto` parameter that you are passing?
The Prisma update method expects the data property to match the structure defined in your Prisma schema.

**Reply by nurul3101** [reply]
@Dsantosjaime I am checking to see if you had a chance to look at my last response.

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem 🙏
